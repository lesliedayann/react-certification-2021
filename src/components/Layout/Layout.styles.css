.container {
  width: 100%;
  min-height: 100%;
}
.main-page-styled{
  background-size: 400% 400%;
  background-position: var(--bg-position);
  transition: background-position 2s ease;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}
.background-light{
  background-image: linear-gradient(
    120deg,
    #eea2a2 0,
    #bbc1bf 19%,
    #57c6e1 42%,
    #b49fda 79%,
    #7ac5d8 100%
  );
}
.background-dark{
  background-image: linear-gradient(
    120deg,
    #423737 0,
    #9fa8a5 19%,
    #454b4c 42%,
    #908a9a 79%,
    #445a5f 100%
  );
}