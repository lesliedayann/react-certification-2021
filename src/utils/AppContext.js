import { createContext, useState } from 'react';

const appContextDefaultValue = {
  search: 'wizeline',
  setSearch: () => {},
  videosData: {},
  setVideosData: () => {},
  videoDetails: {},
  setVideoDetails: () => {},
  darkMode: false,
  setDarkMode: () => {},
};

export const AppContext = createContext(appContextDefaultValue);

export function useAppContextValue() {
  const [search, setSearch] = useState(appContextDefaultValue.search);
  const [videosData, setVideosData] = useState(appContextDefaultValue.videosData);
  const [videoDetails, setVideoDetails] = useState(appContextDefaultValue.videoDetails);
  const [darkMode, setDarkMode] = useState(appContextDefaultValue.darkMode);

  const handleSearchChange = (value) => {
    setSearch(value);
  };
  const handleVideosDataChange = (data) => {
    setVideosData(data);
  };
  const handleVideoDetailsChange = (data) => {
    setVideoDetails(data);
  };
  const handleDarkModeChange = () => {
    setDarkMode(!darkMode);
  };
  return {
    search,
    setSearch: handleSearchChange,
    videosData,
    setVideosData: handleVideosDataChange,
    videoDetails,
    setVideoDetails: handleVideoDetailsChange,
    darkMode,
    setDarkMode: handleDarkModeChange,
  };
}
