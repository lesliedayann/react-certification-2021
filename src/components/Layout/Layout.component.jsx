// import React, { useRef } from 'react';
import React, { useLayoutEffect, useContext, useRef } from 'react';
import { random } from '../../utils/fns';
import { AppContext } from '../../utils/AppContext';
import './Layout.styles.css';

function Layout({ children }) {
  const { darkMode } = useContext(AppContext);
  const mainPage = useRef();

  useLayoutEffect(() => {
    const page = mainPage.current;

    function rotateBackground() {
      const xPercent = random(100);
      const yPercent = random(100);
      page.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
    }
    const intervalId = setInterval(rotateBackground, 3000);
    page.addEventListener('click', rotateBackground);

    return () => {
      clearInterval(intervalId);
      page.removeEventListener('click', rotateBackground);
    };
  }, []);

  return (
    <main
      className={`container main-page-styled ${
        darkMode ? 'background-dark' : 'background-light'
      }`}
      ref={mainPage}
    >
      {children}
    </main>
  );
}

export default Layout;
