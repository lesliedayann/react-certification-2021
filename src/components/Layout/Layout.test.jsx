import React from 'react';
import { mount } from 'enzyme';
import Layout from './Layout.component';
import { AppProvider, AppContext } from '../../utils/AppContext.provider';

describe('<Layout />', () => {
  const wrapper = mount(
    <AppProvider>
      <Layout>
        <p>This is a child</p>
      </Layout>
    </AppProvider>
  );

  test('main container should be rendered correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Render Layout Component', () => {
    expect(wrapper).not.toBeNull();
  });

  test('Render children', () => {
    expect(wrapper.find('p').exists()).toBe(true);
  });
  test('Render background-light', () => {
    expect(wrapper.find('.background-light').exists()).toBe(true);
  });

  test('Render background-dark, darkMode: true', () => {
    const layoutMain = mount(
      <AppContext.Provider value={{ state: { darkMode: true } }}>
        <Layout>
          <p>This is a child</p>
        </Layout>
      </AppContext.Provider>
    );
    expect(layoutMain.find('.background-dark').exists()).toBe(true);
  });
});
