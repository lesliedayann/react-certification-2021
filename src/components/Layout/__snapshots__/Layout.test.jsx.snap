// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> main container should be rendered correctly 1`] = `
<main
  className="container main-page-styled background-light"
>
  <p>
    This is a child
  </p>
</main>
`;
